test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/TestFiles/Input/RazorPages.cshtml

```diff
@@ -0,0 +1,38 @@
+﻿@page
+
+@model NewModel
+@using Microsoft.AspNetCore.Mvc.RazorPages
+
+@functions {
+    public class NewModel : PageModel
+    {
+        public IActionResult OnPost(Customer customer)
+        {
+            return Redirect("~/customers/inlinepagemodels/");
+        }
+
+        public Customer Customer { get; }
+    }
+
+    public class Customer
+    {
+        public string Name { get; set; }
+    }
+}
+
+<h1>New Customer</h1>
+<form method="post" class="form-horizontal" asp-antiforgery="true">
+    <div asp-validation-summary="All" class="text-danger"></div>
+    <div class="form-group">
+        <label asp-for="Customer.Name" class="col-md-2 control-label"></label>
```

Comment:
Also, going on that thought might want to throw in some sort of Razor expression (`@DateTime.Now` equivalent)